File: frameworks/test.web/src/main/java/com/example/scaffold/frameworks/test/web/TestResponse.java
package com.example.scaffold.frameworks.test.web;

import com.jayway.jsonpath.JsonPath;
import net.minidev.json.JSONArray;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import java.util.List;

public class TestResponse {
    private final ResponseEntity<String> response;

    public TestResponse(ResponseEntity<String> response) {
        this.response = response;
    }

    public HttpStatus statusCode() {
        return response.getStatusCode();
    }

    public <T> T value(String jsonPath) {
        return JsonPath.compile(jsonPath).read(response.getBody());
    }

    public Object[] values(String jsonPath) {
        return JsonPath.compile(jsonPath).<JSONArray>read(response.getBody()).toArray();
    }

    public String textBody() {
        return response.getBody();
    }

    public List<String> header(String header) {
        return response.getHeaders().get(header);
    }

    public TestResponseAssert assertStatus(ResponseStatusMatcher matcher) {
        matcher.match(this);
        return new TestResponseAssert(this);
    }
}

File: frameworks/test.web/src/main/kotlin/com/example/scaffold/frameworks/test/web/TestResponse.kt
package com.example.scaffold.frameworks.test.web

import com.jayway.jsonpath.JsonPath
import net.minidev.json.JSONArray
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

class TestResponse(private val response: ResponseEntity<String>) {
    fun statusCode(): HttpStatus {
        return response.statusCode
    }

    fun <T> value(jsonPath: String): T {
        return JsonPath.compile(jsonPath).read(response.body)
    }

    fun values(jsonPath: String): Array<Any> {
        return JsonPath.compile(jsonPath).read<JSONArray>(response.body).toTypedArray()
    }

    fun textBody(): String? {
        return response.body
    }

    fun header(header: String): List<String>? {
        return response.headers[header]
    }

    fun `is`(matcher: ResponseMatcher): TestResponse {
        matcher.match(this)
        return this
    }
}
